gateway/src/apicast/policy/ngx_variable.lua

```diff
@@ -3,17 +3,11 @@ local LinkedList = require('apicast.linked_list')
 local _M = {}
 
 local function context_values()
-  return {
-    uri = ngx.var.uri,
-    host = ngx.var.host,
-    remote_addr = ngx.var.remote_addr,
-    remote_port = ngx.var.remote_port,
-    scheme = ngx.var.scheme,
-    server_addr = ngx.var.server_addr,
-    server_port = ngx.var.server_port,
+  local context = {
     headers = ngx.req.get_headers(),
     http_method = ngx.req.get_method(),
   }
+  return LinkedList.readonly(context, ngx.var)
```

Comment:
I think this is OK, but it concerns me a bit. We were exposing just a few vars and now all of them are.
Can we be sure that nothing confidential is exposed? Maybe we should be more conservative here and select only the few vars that we know are safe? There are many vars: https://nginx.org/en/docs/varindex.html